src/libcore/fmt/mod.rs

```diff
@@ -339,7 +375,37 @@ pub trait Display {
     fn fmt(&self, &mut Formatter) -> Result;
 }
 
-/// Format trait for the `o` character
+/// Format trait for the `o` character.
+///
+/// The `Octal` trait should format its output as a number in base-8.
+///
+/// For more information on formatters, see [the module-level documentation][module].
+///
+/// [module]: ../index.html
+///
+/// # Examples
+///
+/// Implementing `Octal` on a type:
+///
+/// ```
+/// use std::fmt;
+///
+/// struct Length(i32);
+///
+/// impl fmt::Octal for Length {
+///     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+///         let val = self.0;
+///
+///         // convert val to base 8 here in a real implementation
+///
+///         write!(f, "{}", val)
```

Comment:
Can you use `{:o}` to format `val` here? Same with the binary case, you'd just reuse `{:b}` on the inner value. Maybe it's a bit confusing but that's what a real world impl would do I think.